src/api_authentication_app_skeleton/spec/support/app_request.cr

```diff
@@ -0,0 +1,83 @@
+require "http/client"
+
+class AppRequest
+  getter! response
+  @response : HTTP::Client::Response? = nil
+
+  def get(path : String, as user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", user.generate_token
+    end
+
+    request("GET", path, nil, headers)
+  end
+
+  def put(path : String, body : Hash(String, String), as user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", user.generate_token
+    end
+
+    request("PUT", path, body, headers)
+  end
+
+  def post(path : String, body : Hash(String, String), as user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", user.generate_token
+    end
+
+    request("POST", path, body, headers)
+  end
+
+  def delete(path : String, as user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", user.generate_token
+    end
+
+    request("DELETE", path, nil, headers)
+  end
+
+  def response
+    @response.not_nil!    
+  end
+
+  def response_body
+    if response.success?
+      response.body
+    else
+      raise Exception.new(response.body)
+    end
+  end
+
+  def response_json
+    if response_body.empty?
+      ""
+    else
+      JSON.parse(response_body)
+    end
+  end
+
+  private def request(method :  String, path : String, body : Hash(String, String)? = nil, headers : HTTP::Headers? = nil)
+    url = Lucky::RouteHelper.settings.base_uri + path
+    body = body ? hash_to_params(body) : nil
+
+    @response = HTTP::Client.exec(method, url: url, body: body, headers: headers)
+  end
+
+  private def hash_to_params(body : Hash(String, String))
+    body_strings = [] of String
+    body.each do |key, value|
+      body_strings << "#{URI.escape(key)}=#{URI.escape(value)}"
+    end
+    body_strings.join("&")
+  end
+
+  private def hash_to_headers(hash : Hash(String, String))
+    headers = HTTP::Headers.new
+    hash.keys.map { |key| headers.add key, hash[key] }
+    headers
+  end
```

Comment:
I think it would be awesome to extract this to `lucky` itself as `Lucky::Request`. Then in lucky_cli create a `AppRequest < Lucky::Request`. That way if people want they can add their own methods to `AppRequest` and we can make bug fixes in Lucky without people needing to update the generated file. Thoughts?

Reply:
I think it depends a bit on how much time you have. If you are open to it, I think adding this to Lucky before merging this PR would be great. And then this PR could use it.

If you don't have much time then yeah we can make an issue for it and extract it later

Reply:
Need a hand? Happy to help with that migration to `lucky`!